src/Integrations/Integrations/CodeIgniter/V2/CodeIgniterIntegration.php

```diff
@@ -70,7 +70,11 @@ function (SpanData $span) use ($rootSpan, $method, $service) {
                 $span->service = $service;
                 $span->type = Type::WEB_SERVLET;
 
-                $this->load->helper('url');
+                // We took the assumption that all controllers will extend CI_Controller.
+                // But we've at least seen one healthcheck controller not extending it.
+                if ($this->load) {
+                    $this->load->helper('url');
+                }
 
                 if (!array_key_exists(Tag::HTTP_URL, $rootSpan->meta)) {
                     $rootSpan->meta[Tag::HTTP_URL] = \DDTrace\Util\Normalizer::urlSanitize(base_url(uri_string()))
```

Comment:
I would expect this to fail at the call to `base_url()` in case line 76 won't get executed as the `base_url()` function seems to be provided by the `url` helper: https://codeigniter.com/userguide3/helpers/url_helper.html#base_url

Reply:
Thanks a lot. I've regrouped the whole lot in the if in c75e1a80ea427923d66779fffa11e16bdf825e6b.